### ISSUE TITLE
Conversion to PDF fails due to dep on PyQtWebEngine

### ISSUE THREAD

**Issue by parsoj** [linuxserver/docker-calibre-web#154]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
Books successfully convert to PDF, and new PDF format shows up as another format for that book in the UI

## Current Behavior
Convert fails - with the following debug logs:

```
[2021-08-03 23:02:47,606]  INFO {cps.editbooks:1052} converting: book id: 316 from: AZW3 to: PDF
[2021-08-03 23:02:47,611] DEBUG {cps.services.worker:72} Add Task for user: admin: <cps.tasks.convert.TaskConvert object at 0x7fca2c924700>
[2021-08-03 23:02:47,616]  INFO {cps.tasks.convert:102} Book id 316 - target format of .pdf does not exist. Moving forward with convert.
[2021-08-03 23:02:47,800] DEBUG {cps.tasks.convert:207} No write acces to /root/.config/calibre using a temporary dir instead
[2021-08-03 23:02:47,827] DEBUG {cps.tasks.convert:207} 1% Converting input to HTML...
[2021-08-03 23:02:47,872] DEBUG {cps.tasks.convert:207} 
[2021-08-03 23:02:47,875] DEBUG {cps.tasks.convert:222} Failed to import PyQt module: PyQt5.QtWebEngineCore with error: libXdamage.so.1: cannot open shared object file: No such file or directory
[2021-08-03 23:02:47,876] DEBUG {cps.tasks.convert:222} Traceback (most recent call last):
[2021-08-03 23:02:47,877] DEBUG {cps.tasks.convert:222}   File "runpy.py", line 194, in _run_module_as_main
[2021-08-03 23:02:47,878] DEBUG {cps.tasks.convert:222}   File "runpy.py", line 87, in _run_code
[2021-08-03 23:02:47,881] DEBUG {cps.tasks.convert:222}   File "site.py", line 45, in <module>
[2021-08-03 23:02:47,883] DEBUG {cps.tasks.convert:222}   File "site.py", line 41, in main
[2021-08-03 23:02:47,885] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/cli.py", line 401, in main
[2021-08-03 23:02:47,887] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/plumber.py", line 1103, in run
[2021-08-03 23:02:47,888] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/plugins/pdf_output.py", line 151, in specialize_options
[2021-08-03 23:02:47,889] DEBUG {cps.tasks.convert:222} ImportError: cannot import name 'QWebEngineUrlScheme' from 'qt.webengine' (/app/calibre/lib/calibre-extensions/python-lib.bypy.frozen/qt/webengine.pyc)
[2021-08-03 23:02:47,891]  INFO {cps.tasks.convert:140} ebook converter failed with error while converting book
```

installing PyQTWebEngine and libxdamage-dev fixes it for me:

```
docker exec calibre-web apt-get install -y python3-distutils
docker exec calibre-web apt-get install -y python3-apt

docker exec calibre-web apt-get install -y libxdamage-dev
docker exec calibre-web pip install PyQtWebEngine
```

## Steps to Reproduce

1.  run the following docker-compose file:

```
version: "2.1"
services:
  calibre-web:
    image: linuxserver/calibre-web
    container_name: calibre-web
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/Los_Angeles
      - DOCKER_MODS=linuxserver/calibre-web:calibre #optional
      - OAUTHLIB_RELAX_TOKEN_SCOPE=1 #optional
    volumes:
      - ~/Library/Preferences/calibre:/config
      - ~/Dropbox/library/Books:/books
    ports:
      - 8083:8083
    restart: unless-stopped
```

2. open the "Edit Metadata" view for a book. 
3. use the left sidebar to convert the book from AZW3 or EPUB -- to PDF
4. View Tasks page to see failure, and view DEBUG logs to see the above stacktrace reproduced

Quick Fix:
```
docker exec calibre-web apt-get install -y python3-distutils
docker exec calibre-web apt-get install -y python3-apt

docker exec calibre-web apt-get install -y libxdamage-dev
docker exec calibre-web pip install PyQtWebEngine

```

## Environment
**OS: OSX 11.4**
**CPU architecture:** x86_64
**How docker service was installed:**
brew install 

## Command used to create docker container (run/create/compose/screenshot)
docker compose up (using the given compose file)

## Docker logs
Posting a only snippet of the log for privacy reasons
```
[2021-08-03 23:02:47,606]  INFO {cps.editbooks:1052} converting: book id: 316 from: AZW3 to: PDF
[2021-08-03 23:02:47,611] DEBUG {cps.services.worker:72} Add Task for user: admin: <cps.tasks.convert.TaskConvert object at 0x7fca2c924700>
[2021-08-03 23:02:47,616]  INFO {cps.tasks.convert:102} Book id 316 - target format of .pdf does not exist. Moving forward with convert.
[2021-08-03 23:02:47,800] DEBUG {cps.tasks.convert:207} No write acces to /root/.config/calibre using a temporary dir instead
[2021-08-03 23:02:47,827] DEBUG {cps.tasks.convert:207} 1% Converting input to HTML...
[2021-08-03 23:02:47,872] DEBUG {cps.tasks.convert:207} 
[2021-08-03 23:02:47,875] DEBUG {cps.tasks.convert:222} Failed to import PyQt module: PyQt5.QtWebEngineCore with error: libXdamage.so.1: cannot open shared object file: No such file or directory
[2021-08-03 23:02:47,876] DEBUG {cps.tasks.convert:222} Traceback (most recent call last):
[2021-08-03 23:02:47,877] DEBUG {cps.tasks.convert:222}   File "runpy.py", line 194, in _run_module_as_main
[2021-08-03 23:02:47,878] DEBUG {cps.tasks.convert:222}   File "runpy.py", line 87, in _run_code
[2021-08-03 23:02:47,881] DEBUG {cps.tasks.convert:222}   File "site.py", line 45, in <module>
[2021-08-03 23:02:47,883] DEBUG {cps.tasks.convert:222}   File "site.py", line 41, in main
[2021-08-03 23:02:47,885] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/cli.py", line 401, in main
[2021-08-03 23:02:47,887] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/plumber.py", line 1103, in run
[2021-08-03 23:02:47,888] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/plugins/pdf_output.py", line 151, in specialize_options
[2021-08-03 23:02:47,889] DEBUG {cps.tasks.convert:222} ImportError: cannot import name 'QWebEngineUrlScheme' from 'qt.webengine' (/app/calibre/lib/calibre-extensions/python-lib.bypy.frozen/qt/webengine.pyc)
[2021-08-03 23:02:47,891]  INFO {cps.tasks.convert:140} ebook converter failed with error while converting book
```

**Comment by parsoj** [comment]
I'll look into PR-ing a fix that installs PyQtWebEngine and libxdamage-dev to the image, but I'm not sure if that would be considered to "bloated" of an install base for the container. 

I'm not sure at the moment the "minimal dependency tree/dag" will satisfy this in terms of apt or pip packages so I'll just PR the fix I wrote above, and will just be open to suggestions!

**Comment by CHBMB** [comment]
What I've been left wondering is if `PyQtWebEngine` is a necessary or optional dependency, then would it be better to have that stipulated in `requirements.txt` or `optional-requirements.txt` which we use to install every other necessary/optional python package.

https://github.com/linuxserver/docker-calibre-web/blob/master/Dockerfile#L47-L49

Then I realised this might actually be a better dependency to add to the calibre docker mod as that's what is responsible for ebook conversion.

**Comment by CHBMB** [comment]
Just delving into this a bit more, I can reproduce it with an AZW3 file here.  However I managed to get it all working just by installing the missing lib file.

Can you try pulling fresh images and just running

```
docker exec calibre-web apt-get install -y libxdamage1
```

If you confirm that as a fix I'll add it to the docker-mods calibre branch.  It's a lot less bloated than the proposed fix above (which I can confirm also works but I suspect that's as simple as just pulling `libdamage1` as a dependency of the dev package.

**Comment by CHBMB** [comment]
Logs to confirm failure

```
[2021-08-04 21:07:03,804]  INFO {cps.editbooks:1061} converting: book id: 0001 from: AZW3 to: PDF
[2021-08-04 21:07:03,814] DEBUG {cps.services.worker:72} Add Task for user: test: <cps.tasks.convert.TaskConvert object at 0x14cde47f40a0>
[2021-08-04 21:07:04,333]  INFO {cps.tasks.convert:102} Book id 2815 - target format of .pdf does not exist. Moving forward with convert.
[2021-08-04 21:07:04,638] DEBUG {cps.tasks.convert:207} No write acces to /root/.config/calibre using a temporary dir instead
[2021-08-04 21:07:04,693] DEBUG {cps.tasks.convert:207} 1% Converting input to HTML...
[2021-08-04 21:07:04,771] DEBUG {cps.tasks.convert:207} 
[2021-08-04 21:07:04,771] DEBUG {cps.tasks.convert:222} Failed to import PyQt module: PyQt5.QtWebEngineCore with error: libXdamage.so.1: cannot open shared object file: No such file or directory
[2021-08-04 21:07:04,772] DEBUG {cps.tasks.convert:222} Traceback (most recent call last):
[2021-08-04 21:07:04,772] DEBUG {cps.tasks.convert:222}   File "runpy.py", line 194, in _run_module_as_main
[2021-08-04 21:07:04,772] DEBUG {cps.tasks.convert:222}   File "runpy.py", line 87, in _run_code
[2021-08-04 21:07:04,772] DEBUG {cps.tasks.convert:222}   File "site.py", line 45, in <module>
[2021-08-04 21:07:04,772] DEBUG {cps.tasks.convert:222}   File "site.py", line 41, in main
[2021-08-04 21:07:04,772] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/cli.py", line 401, in main
[2021-08-04 21:07:04,773] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/plumber.py", line 1103, in run
[2021-08-04 21:07:04,773] DEBUG {cps.tasks.convert:222}   File "calibre/ebooks/conversion/plugins/pdf_output.py", line 151, in specialize_options
[2021-08-04 21:07:04,773] DEBUG {cps.tasks.convert:222} ImportError: cannot import name 'QWebEngineUrlScheme' from 'qt.webengine' (/app/calibre/lib/calibre-extensions/python-lib.bypy.frozen/qt/webengine.pyc)
[2021-08-04 21:07:04,774]  INFO {cps.tasks.convert:140} ebook converter failed with error while converting book
```

Logs following install `libxdamage`

```
[2021-08-04 21:08:42,187]  INFO {cps.editbooks:1061} converting: book id: 0001 from: AZW3 to: PDF
[2021-08-04 21:08:42,192] DEBUG {cps.services.worker:72} Add Task for user: test: <cps.tasks.convert.TaskConvert object at 0x14cde4885340>
[2021-08-04 21:08:42,836]  INFO {cps.tasks.convert:102} Book id 0001 - target format of .pdf does not exist. Moving forward with convert.
[2021-08-04 21:08:43,179] DEBUG {cps.tasks.convert:207} No write acces to /root/.config/calibre using a temporary dir instead
[2021-08-04 21:08:43,239] DEBUG {cps.tasks.convert:207} 1% Converting input to HTML...
[2021-08-04 21:08:43,401] DEBUG {cps.tasks.convert:207} InputFormatPlugin: MOBI Input running
[2021-08-04 21:08:43,402] DEBUG {cps.tasks.convert:207} on /books/test/test (0001)/test.azw3
[2021-08-04 21:08:43,422] DEBUG {cps.tasks.convert:207} Found KF8 MOBI of type 'standalone'
[2021-08-04 21:08:43,743] DEBUG {cps.tasks.convert:207} KF8 has no metadata Table of Contents
[2021-08-04 21:08:43,754] DEBUG {cps.tasks.convert:207} Parsing all content...
[2021-08-04 21:08:43,808] DEBUG {cps.tasks.convert:207} Forcing text/part0011.html into XHTML namespace
[2021-08-04 21:08:43,811] DEBUG {cps.tasks.convert:207} Forcing text/part0013.html into XHTML namespace
[2021-08-04 21:08:43,813] DEBUG {cps.tasks.convert:207} Forcing text/part0003.html into XHTML namespace
[2021-08-04 21:08:43,817] DEBUG {cps.tasks.convert:207} Forcing text/part0007.html into XHTML namespace
[2021-08-04 21:08:43,821] DEBUG {cps.tasks.convert:207} Forcing text/part0017.html into XHTML namespace
[2021-08-04 21:08:43,823] DEBUG {cps.tasks.convert:207} Forcing text/part0021.html into XHTML namespace
[2021-08-04 21:08:43,825] DEBUG {cps.tasks.convert:207} Forcing text/part0025.html into XHTML namespace
[2021-08-04 21:08:43,828] DEBUG {cps.tasks.convert:207} Forcing text/part0029.html into XHTML namespace
[2021-08-04 21:08:43,846] DEBUG {cps.tasks.convert:207} Forcing text/part0012.html into XHTML namespace
[2021-08-04 21:08:43,850] DEBUG {cps.tasks.convert:207} Forcing text/part0002.html into XHTML namespace
[2021-08-04 21:08:43,852] DEBUG {cps.tasks.convert:207} Forcing text/part0006.html into XHTML namespace
[2021-08-04 21:08:43,854] DEBUG {cps.tasks.convert:207} Forcing text/part0016.html into XHTML namespace
[2021-08-04 21:08:43,856] DEBUG {cps.tasks.convert:207} Forcing text/part0010.html into XHTML namespace
[2021-08-04 21:08:43,857] DEBUG {cps.tasks.convert:207} Forcing text/part0020.html into XHTML namespace
[2021-08-04 21:08:43,859] DEBUG {cps.tasks.convert:207} Forcing text/part0024.html into XHTML namespace
[2021-08-04 21:08:43,864] DEBUG {cps.tasks.convert:207} Forcing text/part0028.html into XHTML namespace
[2021-08-04 21:08:43,879] DEBUG {cps.tasks.convert:207} Forcing text/part0001.html into XHTML namespace
[2021-08-04 21:08:43,881] DEBUG {cps.tasks.convert:207} Forcing text/part0005.html into XHTML namespace
[2021-08-04 21:08:43,882] DEBUG {cps.tasks.convert:207} Forcing text/part0015.html into XHTML namespace
[2021-08-04 21:08:43,884] DEBUG {cps.tasks.convert:207} Forcing text/part0009.html into XHTML namespace
[2021-08-04 21:08:43,886] DEBUG {cps.tasks.convert:207} Forcing text/part0019.html into XHTML namespace
[2021-08-04 21:08:43,889] DEBUG {cps.tasks.convert:207} Forcing text/part0023.html into XHTML namespace
[2021-08-04 21:08:43,892] DEBUG {cps.tasks.convert:207} Forcing text/part0027.html into XHTML namespace
[2021-08-04 21:08:43,910] DEBUG {cps.tasks.convert:207} Forcing text/part0014.html into XHTML namespace
[2021-08-04 21:08:43,913] DEBUG {cps.tasks.convert:207} Forcing text/part0008.html into XHTML namespace
[2021-08-04 21:08:43,915] DEBUG {cps.tasks.convert:207} Forcing text/part0000.html into XHTML namespace
[2021-08-04 21:08:43,917] DEBUG {cps.tasks.convert:207} Forcing text/part0004.html into XHTML namespace
[2021-08-04 21:08:43,918] DEBUG {cps.tasks.convert:207} Forcing text/part0018.html into XHTML namespace
[2021-08-04 21:08:43,920] DEBUG {cps.tasks.convert:207} Forcing text/part0022.html into XHTML namespace
[2021-08-04 21:08:43,923] DEBUG {cps.tasks.convert:207} Forcing text/part0026.html into XHTML namespace
[2021-08-04 21:08:43,926] DEBUG {cps.tasks.convert:207} Selector: Unexpected CHAR. [2:14: =]
[2021-08-04 21:08:43,926] DEBUG {cps.tasks.convert:207} Unexpected token (INVALID, ", 2, 15)
[2021-08-04 21:08:43,926] DEBUG {cps.tasks.convert:207} Unexpected token (PAGE_SYM, @page, 3, 4)
[2021-08-04 21:08:43,926] DEBUG {cps.tasks.convert:207} SelectorList: Invalid Selector: #data data="
[2021-08-04 21:08:43,927] DEBUG {cps.tasks.convert:207} 			@page Section1 
[2021-08-04 21:08:43,928] DEBUG {cps.tasks.convert:207} CSSStyleRule: No start { of style declaration found: '">#data>"' [5:12: ]
[2021-08-04 21:08:43,928] DEBUG {cps.tasks.convert:207} Selector: Unexpected STRING. [5:3: ">#data>"]
[2021-08-04 21:08:43,929] DEBUG {cps.tasks.convert:207} Selector: Invalid or incomplete selector: ">#data>"
[2021-08-04 21:08:43,929] DEBUG {cps.tasks.convert:207} SelectorList: Invalid Selector: ">#data>"
[2021-08-04 21:08:43,929] DEBUG {cps.tasks.convert:207} CSSStyleRule: No style declaration or "}" found: '">#data>"'
[2021-08-04 21:08:43,957] DEBUG {cps.tasks.convert:207} 34% Running transforms on e-book...
[2021-08-04 21:08:44,071] DEBUG {cps.tasks.convert:207} Merging user specified metadata...
[2021-08-04 21:08:44,071] DEBUG {cps.tasks.convert:207} Detecting structure...
[2021-08-04 21:08:44,084] DEBUG {cps.tasks.convert:207} 	Detected chapter: Prologue
[2021-08-04 21:08:44,102] DEBUG {cps.tasks.convert:207} Flattening CSS and remapping font sizes...
[2021-08-04 21:08:47,324] DEBUG {cps.tasks.convert:207} Source base font size is 12.00000pt
[2021-08-04 21:08:47,540] DEBUG {cps.tasks.convert:207} Removing fake margins...
[2021-08-04 21:08:47,727] DEBUG {cps.tasks.convert:207} Removing level p_2 left margin of: 0%
[2021-08-04 21:08:47,728] DEBUG {cps.tasks.convert:207} Removing level p_2 right margin of: 0%
[2021-08-04 21:08:47,824] DEBUG {cps.tasks.convert:207} Cleaning up manifest...
[2021-08-04 21:08:47,824] DEBUG {cps.tasks.convert:207} Trimming unused files from manifest...
[2021-08-04 21:08:47,837] DEBUG {cps.tasks.convert:207} Trimming 'images/00003.jpeg' from manifest
[2021-08-04 21:08:47,837] DEBUG {cps.tasks.convert:207} Creating PDF Output...
[2021-08-04 21:08:47,837] DEBUG {cps.tasks.convert:207} 67% Running PDF Output plugin
[2021-08-04 21:08:47,909] DEBUG {cps.tasks.convert:207} The cover image has an id != "cover". Renaming to work around bug in Nook Color
[2021-08-04 21:08:47,991] DEBUG {cps.tasks.convert:207} 68% Parsed all content for markup transformation
[2021-08-04 21:08:48,072] DEBUG {cps.tasks.convert:207} 70% Completed markup transformation
[2021-08-04 21:08:50,541] DEBUG {cps.tasks.convert:207} 90% Rendered all HTML as PDF
[2021-08-04 21:08:50,542] DEBUG {cps.tasks.convert:207} 91% Added links to PDF content
[2021-08-04 21:08:50,807] DEBUG {cps.tasks.convert:207} Merged 60 instances of DejaVuSans-Bold reducing size from 221.5 KB to 28.5 KB
[2021-08-04 21:08:51,410] DEBUG {cps.tasks.convert:207} Merged 60 instances of DejaVuSans reducing size from 1.2 MB to 45.3 KB
[2021-08-04 21:08:51,428] DEBUG {cps.tasks.convert:207} 100% Updated metadata in PDF
[2021-08-04 21:08:51,523] DEBUG {cps.tasks.convert:207} PDF output written to on /books/test/test (0001)/test.pdf
[2021-08-04 21:08:51,524] DEBUG {cps.tasks.convert:207} Output saved to   on /books/test/test (0001)/test.azw3
[2021-08-04 21:08:51,712] DEBUG {cps.tasks.convert:207} 
[2021-08-04 21:08:51,713] DEBUG {cps.tasks.convert:222} QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-abc'
```

**Comment by CHBMB** [comment]
I made that a whole lot more difficult than it ever should have been.......

**Comment by parsoj** [comment]
hey! I was just starting to dive into how to do a local dev set up for the docker-mods stuff but it looks like you beat me to it! 

I can confirm that just installing libxdamage1 with:
```
docker exec calibre-web apt-get install -y libxdamage1
```

fixes the issue for me - as well!

**Comment by aptalca** [comment]
Fixed in #157
